moped-toolbox/random_project_seeds/index.js

```diff
@@ -0,0 +1,221 @@
+  );
+};
+
+const randomProjPhase = () => ({
+  completion_percentage: 0,
+  completed: false,
+  phase_start: randomDate(),
+  phase_end: randomDate(),
+  phase_id: randomArrElement(lookups.phases),
+  subphase_id: randomArrElement(lookups.subphases),
+});
+
+const randomProjMilestone = () => ({
+  milestone_id: randomArrElement(lookups.milestones),
+  milestone_description: randomString(80),
+  completion_percentage: 0,
+  completed: randomBool(),
+  milestone_start: randomDate(),
+  milestone_end: randomDate(),
+  milestone_estimate: randomDate(),
+});
+
+const randomProjContract = () => ({
+  contractor: randomString(),
+  contract_number: randomString(),
+  description: randomString(80),
+});
+
+const randomProjPartner = () => ({
+  partner_name: randomString(),
+  entity_id: randomArrElement(lookups.entities),
+});
+
+const randomProjectType = () => ({
+  project_type_id: randomArrElement(lookups.project_types),
+});
+
+const randomNote = () => ({
+  project_note: randomString(200),
+  project_note_type: randomArrElement(lookups.project_note_types),
+  added_by_user_id: randomArrElement(lookups.users),
+});
+
+const randomTag = () => ({
+  tag_id: randomArrElement(lookups.proj_tags),
+});
+
+const randomPersonnel = () => ({
+  user_id: randomArrElement(lookups.users),
+  notes: randomString(80),
+  moped_proj_personnel_roles: {
+    data: [...new Array(settings.roles_per_personnel)].map((_) => ({
+      project_role_id: randomArrElement(lookups.proj_roles),
+    })),
+  },
+});
+
+const randomProjFunding = () => ({
+  funding_source_id: randomArrElement(lookups.fund_sources),
+  funding_program_id: randomArrElement(lookups.fund_programs),
+  funding_amount: randomInteger(4),
+  funding_description: randomString(80),
+  funding_status_id: randomArrElement(lookups.fund_statuses),
+  added_by: randomArrElement(lookups.users),
+  fund: { fund_id: "8400", fund_name: "GCP-TPSD GENERAL IMPROVEMENTS" },
+  dept_unit: {
+    dept: "2507",
+    unit: "9024",
+    dept_id: "180775091",
+    dept_unit_id: "359318059",
+    unit_long_name: "Pay Station Parking Meter Replacement",
+    unit_short_name: "8400-2507",
+    dept_unit_status: "Active",
+  },
+});
+
+const randomProjComponent = () => ({
+  component_id: randomArrElement(lookups.components),
+  name: randomString(),
+  description: randomString(),
+});
+
+const randomProject = () => ({
+  project_name: randomString(),
+  project_description: randomString(),
+  project_lead_id: randomArrElement(lookups.entities),
+  project_sponsor: randomArrElement(lookups.entities),
+  project_website: "https://data.mobility.austin.gov",
+  added_by: randomArrElement(lookups.users),
+  knack_project_id: randomString(),
+  work_assignment_id: randomString(),
+  interim_project_id: randomInteger(3),
+  ecapris_subproject_id: 7333.001,
+  moped_proj_partners: {
+    data: [...new Array(settings.partners)].map((_) => randomProjPartner()),
```

Comment:
this `[...new Array(int)]` pattern just fills an array with an arbitrary number of elements to map against.